zip-0207.rst

```diff
@@ -43,243 +80,129 @@ Specification
-
-================================== ======== ======== ========
-          Example height           Stream A Stream B Stream C
-================================== ======== ======== ========
-     ``AddressChangeInterval - 2``    A0
-     ``AddressChangeInterval - 1``    A0
-     ``AddressChangeInterval``        A1       B0       C0
-     ``AddressChangeInterval + 1``    A1       B0       C0
-             \...
-   ``2*AddressChangeInterval - 2``    A1       B0       C0
-   ``2*AddressChangeInterval - 1``    A1       B0       C0
-   ``2*AddressChangeInterval``        A2                C1
-   ``2*AddressChangeInterval + 1``    A2                C1
-             \...
-``PostBlossomHalvingInterval - 2``    A2                C1
-``PostBlossomHalvingInterval - 1``    A2                C1
-``PostBlossomHalvingInterval``                          C2
-``PostBlossomHalvingInterval + 1``                      C2
-================================== ======== ======== ========
-
-Note that this is not intended to align with the end of a pre-Blossom Founders' Reward address period (as
-defined by ``FounderAddressChangeInterval`` in [#original-fr-consensus-rule]_). There will be a shortened
-Founders' Reward address period prior to Blossom activation.
+A funding stream is defined by a block subsidy fraction (represented as a
+numerator and a denominator), a start height (inclusive), and an end height
+(exclusive).
+
+By defining the issuance as a proportion of the total block subsidy, rather
+than absolute zatoshis, this ZIP dovetails with any changes to both block
+target spacing and issuance-per-block rates. Such a change occurred at the
+Blossom network upgrade, for example. [#zip-0208]_
+
+The value of a funding stream at a given block height is defined as:
+
+.. math::
+
+    \mathsf{FundingStream[FUND].Value}(\mathsf{height}) =
+        \mathsf{floor}\left(
+            \frac{\mathsf{BlockSubsidy}(\mathsf{height}) \,\cdot\, \mathsf{FundingStream[FUND].ValueNumerator}}{\mathsf{FundingStream[FUND].ValueDenominator}}
+        \right)
+
+An active funding stream at a given block height is defined as a funding
+stream for which the block height is less than its end height, but not less
+than its start height.
+
+Each funding stream has an associated sequence of recipient addresses,
+each of which MUST be either a transparent P2SH address or a Sapling address.
+
+Each address is used for at most 1/48th of a halving interval, creating a
+roughly-monthly sequence of funding periods. The address to be used for a
+given block height is defined as follows:
+
+.. math::
+
+    \begin{eqnarray*}
+        \mathsf{AddressChangeInterval} &=& \mathsf{PostBlossomHalvingInterval} / 48 \\
+        \mathsf{AddressPeriod}(\mathsf{height}) &=&
+            \mathsf{floor}\left(
+                {\small\frac{\mathsf{height} + \mathsf{PostBlossomHalvingInterval} - \mathsf{HeightForHalving}(1)}{\mathsf{AddressChangeInterval}}}
```

Comment:
Yeah I don't see why that's necessary. It doesn't make any difference because only differences between two outputs of AddressPeriod are used in the computation of FundingStream[Fund].AddressIndex(height). I guess a C++ implementation would have to be more careful to implement `floor` correctly, since without the addition of PostBlossomHalvingInterval, the AddressPeriod could be negative before the first halving. (This cannot happen on mainnet with the planned activation height, but can happen on testnet.)

Reply:
@str4d, do you think we should change this, or not?